plugins/all-modules-page/src/main/kotlin/ExternalModuleLinkResolver.kt

```diff
@@ -52,15 +51,19 @@ class DefaultExternalModuleLinkResolver(val context: DokkaContext) : ExternalMod
         }
 
     override fun resolve(dri: DRI, fileContext: File): String? {
-        val absoluteLink = elps.mapNotNull { it.resolve(dri) }.firstOrNull() ?: return null
-        val modulePath = context.configuration.outputDir.absolutePath.split(File.separator)
-        val contextPath = fileContext.absolutePath.split(File.separator)
-        val commonPathElements = modulePath.zip(contextPath)
+        val resolvedLinks = elps.mapNotNull { it.resolve(dri)?.removePrefix("file:") }
+        val modulePath = context.configuration.outputDir.absolutePath
+        val validLink = resolvedLinks.firstOrNull {
+            val absolutePath = File(modulePath + it)
+            absolutePath.isFile
+        } ?: return null
```

Comment:
Will it not work for the non-local external `package-list` at all? See the example with the `serialization` library from [External documentation links configuration](https://kotlinlang.org/docs/dokka-gradle.html#external-documentation-links-configuration). Can you fix it, please?

It is nice to fix the issue at least for cross-module links.
A check of existing external URLs online can be excessive and Dokka can be run without access to the Internet. 

Also, I think the fix should check links only if `resolvedLinks` has more than one link. What do you think? It can help avoid unexpected behaviour with resolving cross-module links.
This fix is local and temporary. We have a long-term plan #3368.